cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -287,6 +294,78 @@ public void getAllApps(HttpRequest request, HttpResponder responder,
     }
   }
 
+  /**
+   * Retrieves all namespace source control metadata for applications within the specified namespace.
+   *
+   * @param request      The HTTP request containing parameters for retrieving metadata.
+   * @param responder    The HTTP responder for sending the response.
+   * @param namespaceId  The ID of the namespace for which to retrieve metadata.
+   * @param pageToken    A token for paginating through results.
+   * @param pageSize     The size of each page for pagination.
+   * @param sortOrder    The sort order for the metadata.
+   * @param sortOrderOn  The field on which to sort the metadata.
+   * @param filter       A filter string for filtering the metadata.
+   *                     filter query format - "name=&lt;name-filter&gt; AND syncStatus=&lt;SYNCED/UNSYNCED&gt;".
+   * @throws Exception If an error occurs during the metadata retrieval process.
+   */
+  @GET
+  @Path("/sourcecontrol/apps")
+  public void getAllNamespaceSourceControlMetadata(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespaceId,
+      @QueryParam("pageToken") String pageToken,
+      @QueryParam("pageSize") Integer pageSize,
+      @QueryParam("sortOrder") SortOrder sortOrder,
+      @QueryParam("sortOrderOn") SortBy sortOrderOn,
+      @QueryParam("filter") String filter
+  ) throws Exception {
+    validateNamespace(namespaceId);
+
+    JsonPaginatedListResponder.respond(GSON, responder, APP_LIST_PAGINATED_KEY,
+        jsonListResponder -> {
+          AtomicReference<SourceControlMetadataRecord> lastRecord = new AtomicReference<>(null);
+          ScanSourceControlMetadataRequest scanRequest = SourceControlMetadataHelper.getScmStatusScanRequest(
+              namespaceId,
+              pageToken, pageSize, sortOrder, sortOrderOn, filter);
+          boolean pageLimitReached = false;
+          try {
+            pageLimitReached = applicationLifecycleService.scanSourceControlMetadata(
+                scanRequest, 20,
+                scmMetaRecord -> {
+                  SourceControlMetadataDetail detail = new SourceControlMetadataDetail(
```

Comment:
I think we should just send `SourceControlMeta`. UI can decide what to show

Reply:
SourceControlMeta doesn't have the file name. It has file hash, commitId, LastSynced, and syncStatus. Should I add another property 'name'?

Reply:
Oh right. In that case this is ok. Please add javadoc in the class to explain what is the usecase

Reply:
Sure. Added in SourceControlMetadataDetail class